napari/_vispy/layers/labels.py

```diff
@@ -68,8 +89,14 @@
     // we get a different value:
     // - if it's the empty key, exit;
     // - otherwise, it's a hash collision: continue searching
-    while ((abs(found - t) > 1e-8) && (abs(found - empty) > 1e-8)) {
-        t = t + 1;
+    float initial_t = t;
+    int count = 0;
+    while ((abs(found - initial_t) > 1e-8) && (abs(found - empty) > 1e-8)) {
```

Comment:
Sorry if this is a dumb question, I mostly forgot how this code works... But is it ok that we have `1e-8` epsilon in this comparison here, or should we go lower?

Reply:
the difference between keys should be 1. So 1e-8 is really strict.

Reply:
Ah right, this is just keys comparison, my bad!